src/Service/Parsers/EdmModelBuilder.cs

```diff
@@ -150,6 +102,42 @@ SourceDefinition sourceDefinition
             return this;
         }
 
+        /// <summary>
+        /// Given the system type, returns the corresponding primitive type kind.
+        /// </summary>
+        /// <param name="columnSystemType">Type of the column.</param>
+        /// <returns>EdmPrimitiveTypeKind</returns>
+        /// <exception cref="ArgumentException">Throws when the column</exception>
+        private static EdmPrimitiveTypeKind GetEdmPrimitiveTypeFromSystemType(Type columnSystemType)
```

Comment:
Suggestion: we can move this function to the DbTypeHelper class (could be renamed to a more generic TypeHelper class as already done by Sean in his PR for openapi doc).